homeassistant/components/climate/__init__.py

```diff
@@ -644,17 +650,75 @@ async def async_turn_on(self) -> None:
             if mode not in self.hvac_modes:
                 continue
             await self.async_set_hvac_mode(mode)
-            break
+            return
+
+        raise NotImplementedError
+
+    @property
+    def _is_turn_on_supported(self) -> bool:
+        if (
+            type(self).async_turn_on is not ClimateEntity.async_turn_on
+            or type(self).turn_on is not ClimateEntity.turn_on
+        ):
+            return True
+
+        if len(self.hvac_modes) == 2 and HVACMode.OFF in self.hvac_modes:
+            return True
+
+        for mode in (HVACMode.HEAT_COOL, HVACMode.HEAT, HVACMode.COOL):
+            if mode in self.hvac_modes:
+                return True
+
+        return False
```

Comment:
Instead of adding this, and the corresponding helper to detect if turn off is supported, I'd suggest we wait for https://github.com/home-assistant/core/pull/101673 and then rebase this PR to use the new feature flags. Does that seem OK @arturpragacz ?

Reply:
What do you think about doing the reverse, that is merge this first and then use those helpers for the checking in https://github.com/home-assistant/core/pull/101673 ?

Reply:
That would make no sense as we should use the feature flags so I would then have to remove these checks in the other PR. 
Anyhow it's shortly finished so you don't need to wait long to finalize. 👍

Reply:
@gjohansson-ST You would use exactly those checks to set the flags for the back compat. And then use the flags themselves for everything else.